src/hyp3_isce2/burst.py

```diff
@@ -0,0 +1,238 @@
+            content = copy.deepcopy(elem.find('content'))
+            content.tag = 'product'
+            setattr(self, files[name], content)
+            setattr(self, f'{files[name]}_name', elem.attrib['source_filename'])
+
+        file_paths = [x.attrib['href'] for x in self.manifest.findall('.//fileLocation')]
+        pattern = f'^./measurement/s1.*{self.swath.lower()}.*{self.polarization.lower()}.*.tiff$'
+        self.measurement_name = [Path(x).name for x in file_paths if re.search(pattern, x)][0]
+
+        self.gcp_df = self.create_gcp_df()
+        self.footprint = self.create_geometry(self.gcp_df)[0]
+        self.orbit_direction = self.manifest.findtext('.//{*}pass').lower()
+
+    @staticmethod
+    def reformat_gcp(point):
+        attribs = ['line', 'pixel', 'latitude', 'longitude', 'height']
+        values = {}
+        for attrib in attribs:
+            values[attrib] = float(point.find(attrib).text)
+        return values
+
+    def create_gcp_df(self):
+        points = self.annotation.findall('.//{*}geolocationGridPoint')
+        gcp_df = pd.DataFrame([self.reformat_gcp(x) for x in points])
+        gcp_df = gcp_df.sort_values(['line', 'pixel']).reset_index(drop=True)
+        return gcp_df
+
+    def create_geometry(self, gcp_df):
+        burst_index = self.burst_number - 1
+        lines = int(self.annotation.findtext('.//{*}linesPerBurst'))
+        first_line = gcp_df.loc[gcp_df['line'] == burst_index * lines, ['longitude', 'latitude']]
+        second_line = gcp_df.loc[gcp_df['line'] == (burst_index + 1) * lines, ['longitude', 'latitude']]
+        x1 = first_line['longitude'].tolist()
+        y1 = first_line['latitude'].tolist()
+        x2 = second_line['longitude'].tolist()
+        y2 = second_line['latitude'].tolist()
+        x2.reverse()
+        y2.reverse()
+        x = x1 + x2
+        y = y1 + y2
+        footprint = geometry.Polygon(zip(x, y))
+        centroid = tuple([x[0] for x in footprint.centroid.xy])
+        return footprint, footprint.bounds, centroid
+
+
+def create_burst_request(params: BurstParams, content: str) -> dict:
+    """
+    Syntax: www.API-URL.com/<granule>/<subswath>/<pol>/<burst_number>.(xml|tiff)
+    """
+    filetypes = {'metadata': 'xml', 'geotiff': 'tiff'}
+    exstension = filetypes[content]
+    burst_number_zero_indexed = params.burst_number - 1
+    url = f'{URL}/{params.granule}/{params.swath}/{params.polarization}/{burst_number_zero_indexed}.{exstension}'
+    return {'url': url}
+
+
+def wait_for_extractor(response: requests.Response, sleep_time: int = 15) -> bool:
+    if response.status_code == 202:
+        time.sleep(15)
+        return False
+
+    response.raise_for_status()
+    return True
+
+
+def download_from_extractor(asf_session: requests.Session, burst_params: BurstParams, content: str):
+    burst_request = create_burst_request(burst_params, content=content)
+    burst_request['cookies'] = {'asf-urs': asf_session.cookies['asf-urs']}
```

Comment:
I would suggest renaming `create_burst_request` to something like `get_burst_request_url` and have the function return `str` rather than `dict`, then do something like:

```suggestion
    burst_request = {
        'url': create_burst_request(burst_params, content=content),
        'cookies': {'asf-urs': asf_session.cookies['asf-urs']}
    }
```